nirc_ehr/test/src/org.labkey.test/tests.nirc_ehr/NIRC_EHRTest.java

```diff
@@ -230,6 +261,77 @@ protected String getAnimalHistoryPath()
         return "/ehr/" + PROJECT_NAME + "/animalHistory.view?";
     }
 
+    @Test
+    public void testArrivalForm()
+    {
+        String arrivedAnimal = "30905";
+        LocalDateTime now = LocalDateTime.now();
+
+        gotoEnterData();
+        waitAndClickAndWait(Locator.linkWithText("Arrivals"));
+        lockForm();
+
+        Ext4GridRef arrivals = _helper.getExt4GridForFormSection("Arrivals");
+        _helper.addRecordToGrid(arrivals);
+        arrivals.setGridCellJS(1, "date", now.minusDays(1).format(DateTimeFormatter.ofPattern(DATE_TIME_FORMAT_STRING)));
+        arrivals.setGridCell(1, "Id", arrivedAnimal);
+        arrivals.setGridCell(1, "cage", "C1");
+        arrivals.setGridCell(1, "project", "640991");
+        arrivals.setGridCell(1, "arrivalProtocol", "dummyprotocol");
+        arrivals.setGridCellJS(1, "Id/demographics/birth", now.minusDays(7).format(DateTimeFormatter.ofPattern(DATE_TIME_FORMAT_STRING)));
+        submitFinal();
+
+        goToSchemaBrowser();
+        DataRegionTable table = viewQueryData("study", "arrival");
+        table.setFilter("Id", "Equals", arrivedAnimal);
+        CustomizeView view = table.openCustomizeGrid();
+        view.addColumn("cage");
+        view.addColumn("project");
+        view.addColumn("arrivalProtocol");
+        view.applyCustomView();
+        Assert.assertEquals("Invalid Arrival record", Arrays.asList(arrivedAnimal), table.getRowDataAsText(0, "Id"));
+        Assert.assertEquals("Invalid Arrival record", Arrays.asList("C1"), table.getRowDataAsText(0, "cage"));
+        Assert.assertEquals("Invalid Arrival record", Arrays.asList("640991"), table.getRowDataAsText(0, "project"));
+        Assert.assertEquals("Invalid Arrival record", Arrays.asList("dummyprotocol"), table.getRowDataAsText(0, "arrivalProtocol"));
+
+        verifyRowCreated("study", "birth", arrivedAnimal, 1);
+        verifyRowCreated("study", "assignment", arrivedAnimal, 1);
+        verifyRowCreated("study", "protocolAssignment", arrivedAnimal, 1);
+        verifyRowCreated("study", "housing", arrivedAnimal, 1);
```

Comment:
Also, verify a record is created in study.demographics

Reply:
Good suggestion